src/components/SuggestionsList/SuggestionsList.tsx

```diff
@@ -0,0 +1,155 @@
+import Spinner from '../Spinner';
+import Text, { TextSize, TextColor } from '../Text';
+import Callout from '../Callout';
+import Block, { GutterSize } from '../Block';
+import './SuggestionsList.css';
+
+export interface SuggestionsListProps extends NestableBaseComponentProps {
+  /**
+   * Flag to indicate that a search is in flight
+   * @default false
+   */
+  isLoading?: boolean;
+  /**
+   * Text to show next to the loading spinner.
+   */
+  loadingText: string;
+  /**
+   * Text to show when the results are empty for the active search.
+   */
+  noResultsText: string;
+  /**
+   * The groups of items to render.
+   */
+  groupedItems?: SuggestionItemGroupProps[];
+  /**
+   * The index of current keyboard selection.
+   */
+  selectedId?: string | number;
+  /**
+   * The active search that produced the current state.
+   */
+  searchText: string;
+  /**
+   * Called when an item in a group is selected by clicking.
+   */
+  onItemSelected(item: SuggestionItem): void;
+  /**
+   * The target that the Callout should try to position itself based on.
+   */
+  target: HTMLElement;
+}
+
+interface SuggestionsListWithResultsProps extends SuggestionsListProps {
+  groupedItems: SuggestionItemGroupProps[];
+}
+
+export interface SuggestionItemGroupProps {
+  title: string;
+  items: SuggestionItem[];
+}
+
+export interface SuggestionsListState {
+  hoveredId: string | number | null;
+}
+
+const hasResults = (props: SuggestionsListProps): props is SuggestionsListWithResultsProps => {
+  return !!props.groupedItems && props.groupedItems.length > 0;
+};
+
+const withResultsClass = 'y-suggestions-list--with-results';
+const withStatusClass = 'y-suggestions-list--with-status';
+
+/**
+ * A `SuggestionsList` displays a list of search results in a dropdown.
+ */
+export default class SuggestionsList
+  extends React.PureComponent<SuggestionsListProps, SuggestionsListState> {
+
+  public state: SuggestionsListState = { hoveredId: null };
+
+  public render() {
+    const status = this.getSearchStatus();
+    const results = this.getGroupedResults();
+
+    const classNames = ['y-suggestions-list'];
+    if (status) {
+      classNames.push(withStatusClass);
+    }
+    if (results) {
+      classNames.push(withResultsClass);
+    }
+
+    return (
+      <Callout target={this.props.target} className={classNames.join(' ')} isBeakVisible={false}>
+        {results}
+        {status}
+      </Callout>);
+  }
+
+  private getGroupedResults() {
+    if (!hasResults(this.props)) {
+      return null;
+    }
+    return (
+      <ul className="y-suggestions-list--results">
+        {this.props.groupedItems.map(this.getGroupResults)}
+      </ul>);
+  }
+
+  private getGroupResults = (group: SuggestionItemGroupProps) => {
+    const items = group.items.map(this.getResultItem);
+    return (
+      <li key={group.title}>
+        <ul onMouseOut={this.onMouseOut}>
+          {items}
+        </ul>
+      </li>);
```

Comment:
Can you generate an example screenshot with multiple groups of results with their corresponding titles? I only see a single group with no title in the screenshot.

Reply:
titles has been punted on - we wont be supporting it until we do the grouped search results and who knows when that'll ever be